lib/sms/sms.c

```diff
@@ -227,6 +247,12 @@ static int sms_init(void)
 	/* Check if one SMS client has already been registered. */
 	ret = nrf_modem_at_cmd(sms_buf_tmp, SMS_BUF_TMP_LEN, AT_SMS_SUBSCRIBER_READ);
 	if (ret) {
+#if CONFIG_SMS_CMS_STRERROR
```

Comment:
the difference is that the latter will also evaluate to true for explicitly setting `CONFIG_SMS_CMS_STRERROR=n`

Reply:
I can't reproduce that behavior with `#if defined` and setting the config to `n`. I think I checked this long time ago also because I thought it would behave the way you said. I don't know if Kconfig's differ somehow from the definitions done with `#define`

Reply:
In nrf repository, we seem to have 1388 times `#if defined(CONFIG` and 481 times `#if CONFIG`. I can't remember who told be to use `defined` in these.